pkg/sql/opt/exec/execbuilder/tests/local/local_test: TestExecBuild_lookup_join_limit failed

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_lookup_join_limit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12358539?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12358539?buildTab=artifacts#/) on master @ [c2ffc96f93f7a0dac9799788a042f962148e8af0](https://github.com/cockroachdb/cockroach/commits/c2ffc96f93f7a0dac9799788a042f962148e8af0):


```
                │   │ estimated max sql temp disk usage: 0 B
                │   │ table: a@a_pkey
                │   │
                │   └── • scan
                │         nodes: <hidden>
                │         regions: <hidden>
                │         actual row count: 1
                │         KV time: 0µs
                │         KV contention time: 0µs
                │         KV rows decoded: 1
                │         KV pairs read: 2
                │         KV bytes read: 8 B
                │         KV gRPC calls: 1
                │         missing stats
                │         table: a@a_y_idx
                │         spans: [/1 - /1]
                │
                └── • index join (streamer)
                    │ nodes: <hidden>
                    │ regions: <hidden>
                    │ actual row count: 0
                    │ KV time: 0µs
                    │ KV contention time: 0µs
                    │ KV rows decoded: 0
                    │ KV bytes read: 0 B
                    │ KV gRPC calls: 0
                    │ estimated max memory allocated: 0 B
                    │ estimated max sql temp disk usage: 0 B
                    │ table: a@a_pkey
                    │
                    └── • scan
                          nodes: <hidden>
                          regions: <hidden>
                          actual row count: 0
                          KV time: 0µs
                          KV contention time: 0µs
                          KV rows decoded: 0
                          KV bytes read: 0 B
                          KV gRPC calls: 0
                          missing stats
                          table: a@a_y_idx
                          spans: [/2 - /2]
            
[00:44:13] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12423/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit with config local: 5 tests, 1 failures
    logic.go:3998: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12423/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit:235: error while processing
    logic.go:3998: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/12423/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit:235: too many errors encountered, skipping the rest of the input
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/00379fe61763d6e4fd7f372ff2fcf79a/logTestExecBuild_lookup_join_limit2351652614
--- FAIL: TestExecBuild_lookup_join_limit (2.52s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_lookup_join_limit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-32750

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_lookup_join_limit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12359078?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12359078?buildTab=artifacts#/) on master @ [f32937e866d067ff8f026a983506ff00fbd52427](https://github.com/cockroachdb/cockroach/commits/f32937e866d067ff8f026a983506ff00fbd52427):


```
                │   │ estimated max sql temp disk usage: 0 B
                │   │ table: a@a_pkey
                │   │
                │   └── • scan
                │         nodes: <hidden>
                │         regions: <hidden>
                │         actual row count: 1
                │         KV time: 0µs
                │         KV contention time: 0µs
                │         KV rows decoded: 1
                │         KV pairs read: 2
                │         KV bytes read: 8 B
                │         KV gRPC calls: 1
                │         missing stats
                │         table: a@a_y_idx
                │         spans: [/1 - /1]
                │
                └── • index join (streamer)
                    │ nodes: <hidden>
                    │ regions: <hidden>
                    │ actual row count: 0
                    │ KV time: 0µs
                    │ KV contention time: 0µs
                    │ KV rows decoded: 0
                    │ KV bytes read: 0 B
                    │ KV gRPC calls: 0
                    │ estimated max memory allocated: 0 B
                    │ estimated max sql temp disk usage: 0 B
                    │ table: a@a_pkey
                    │
                    └── • scan
                          nodes: <hidden>
                          regions: <hidden>
                          actual row count: 0
                          KV time: 0µs
                          KV contention time: 0µs
                          KV rows decoded: 0
                          KV bytes read: 0 B
                          KV gRPC calls: 0
                          missing stats
                          table: a@a_y_idx
                          spans: [/2 - /2]
            
[05:28:52] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/13718/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit with config local: 5 tests, 1 failures
    logic.go:3998: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/13718/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit:235: error while processing
    logic.go:3998: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/13718/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit:235: too many errors encountered, skipping the rest of the input
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/00379fe61763d6e4fd7f372ff2fcf79a/logTestExecBuild_lookup_join_limit3377887317
--- FAIL: TestExecBuild_lookup_join_limit (1.16s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_lookup_join_limit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_lookup_join_limit [failed](https://tanzanite.cluster.engflow.com/invocation/4554492b-d778-4517-93b5-6da66c531a9f) on master @ [2205334339583cea57db33e3c0d992f672e660e8](https://github.com/cockroachdb/cockroach/commits/2205334339583cea57db33e3c0d992f672e660e8):


```
                │   │ estimated max sql temp disk usage: 0 B
                │   │ table: a@a_pkey
                │   │
                │   └── • scan
                │         nodes: <hidden>
                │         regions: <hidden>
                │         actual row count: 1
                │         KV time: 0µs
                │         KV contention time: 0µs
                │         KV rows decoded: 1
                │         KV pairs read: 2
                │         KV bytes read: 8 B
                │         KV gRPC calls: 1
                │         missing stats
                │         table: a@a_y_idx
                │         spans: [/1 - /1]
                │
                └── • index join (streamer)
                    │ nodes: <hidden>
                    │ regions: <hidden>
                    │ actual row count: 0
                    │ KV time: 0µs
                    │ KV contention time: 0µs
                    │ KV rows decoded: 0
                    │ KV bytes read: 0 B
                    │ KV gRPC calls: 0
                    │ estimated max memory allocated: 0 B
                    │ estimated max sql temp disk usage: 0 B
                    │ table: a@a_pkey
                    │
                    └── • scan
                          nodes: <hidden>
                          regions: <hidden>
                          actual row count: 0
                          KV time: 0µs
                          KV contention time: 0µs
                          KV rows decoded: 0
                          KV bytes read: 0 B
                          KV gRPC calls: 0
                          missing stats
                          table: a@a_y_idx
                          spans: [/2 - /2]
            
[08:35:18] --- done: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit with config local: 5 tests, 1 failures
    logic.go:3998: 
        /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit:235: error while processing
    logic.go:3998: /var/lib/engflow/worker/work/2/exec/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit:235: too many errors encountered, skipping the rest of the input
    panic.go:523: -- test log scope end --
test logs left over in: outputs.zip/logTestExecBuild_lookup_join_limit3293726706
--- FAIL: TestExecBuild_lookup_join_limit (0.98s)
```
<p>Parameters: <code>attempt=1</code>
, <code>run=22</code>
, <code>shard=43</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_lookup_join_limit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_lookup_join_limit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12367986?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/12367986?buildTab=artifacts#/) on master @ [29dee1472fcd518f1a7f5d97249cd7c2e6f5f923](https://github.com/cockroachdb/cockroach/commits/29dee1472fcd518f1a7f5d97249cd7c2e6f5f923):


```
                │   │ estimated max sql temp disk usage: 0 B
                │   │ table: a@a_pkey
                │   │
                │   └── • scan
                │         nodes: <hidden>
                │         regions: <hidden>
                │         actual row count: 1
                │         KV time: 0µs
                │         KV contention time: 0µs
                │         KV rows decoded: 1
                │         KV pairs read: 2
                │         KV bytes read: 8 B
                │         KV gRPC calls: 1
                │         missing stats
                │         table: a@a_y_idx
                │         spans: [/1 - /1]
                │
                └── • index join (streamer)
                    │ nodes: <hidden>
                    │ regions: <hidden>
                    │ actual row count: 0
                    │ KV time: 0µs
                    │ KV contention time: 0µs
                    │ KV rows decoded: 0
                    │ KV bytes read: 0 B
                    │ KV gRPC calls: 0
                    │ estimated max memory allocated: 0 B
                    │ estimated max sql temp disk usage: 0 B
                    │ table: a@a_pkey
                    │
                    └── • scan
                          nodes: <hidden>
                          regions: <hidden>
                          actual row count: 0
                          KV time: 0µs
                          KV contention time: 0µs
                          KV rows decoded: 0
                          KV bytes read: 0 B
                          KV gRPC calls: 0
                          missing stats
                          table: a@a_y_idx
                          spans: [/2 - /2]
            
[16:09:18] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11439/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit with config local: 5 tests, 1 failures
    logic.go:3998: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11439/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit:235: error while processing
    logic.go:3998: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/11439/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit:235: too many errors encountered, skipping the rest of the input
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/00379fe61763d6e4fd7f372ff2fcf79a/logTestExecBuild_lookup_join_limit3990326602
--- FAIL: TestExecBuild_lookup_join_limit (1.19s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_lookup_join_limit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/opt/exec/execbuilder/tests/local/local_test.TestExecBuild_lookup_join_limit [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12368140?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/12368140?buildTab=artifacts#/) on master @ [5a38b5792ad00cb953ad8e0d297f867f2b94aaf9](https://github.com/cockroachdb/cockroach/commits/5a38b5792ad00cb953ad8e0d297f867f2b94aaf9):


```
                │   │ estimated max sql temp disk usage: 0 B
                │   │ table: a@a_pkey
                │   │
                │   └── • scan
                │         nodes: <hidden>
                │         regions: <hidden>
                │         actual row count: 1
                │         KV time: 0µs
                │         KV contention time: 0µs
                │         KV rows decoded: 1
                │         KV pairs read: 2
                │         KV bytes read: 8 B
                │         KV gRPC calls: 1
                │         missing stats
                │         table: a@a_y_idx
                │         spans: [/1 - /1]
                │
                └── • index join (streamer)
                    │ nodes: <hidden>
                    │ regions: <hidden>
                    │ actual row count: 0
                    │ KV time: 0µs
                    │ KV contention time: 0µs
                    │ KV rows decoded: 0
                    │ KV bytes read: 0 B
                    │ KV gRPC calls: 0
                    │ estimated max memory allocated: 0 B
                    │ estimated max sql temp disk usage: 0 B
                    │ table: a@a_pkey
                    │
                    └── • scan
                          nodes: <hidden>
                          regions: <hidden>
                          actual row count: 0
                          KV time: 0µs
                          KV contention time: 0µs
                          KV rows decoded: 0
                          KV bytes read: 0 B
                          KV gRPC calls: 0
                          missing stats
                          table: a@a_y_idx
                          spans: [/2 - /2]
            
[18:33:44] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3053/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit with config local: 5 tests, 1 failures
    logic.go:3998: 
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3053/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit:235: error while processing
    logic.go:3998: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3053/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/opt/exec/execbuilder/tests/local/local_test_/local_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit:235: too many errors encountered, skipping the rest of the input
    panic.go:523: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0f5ae97386d181c4ea2f7c775e9c3a67/logTestExecBuild_lookup_join_limit3451338289
--- FAIL: TestExecBuild_lookup_join_limit (1.41s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestExecBuild_lookup_join_limit.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit
+++ b/pkg/sql/opt/exec/execbuilder/testdata/lookup_join_limit
@@ -4,7 +4,7 @@
 # rows eagerly in the presence of limit hints.
 
 statement ok
-CREATE TABLE a (x INT PRIMARY KEY, y INT, z INT, INDEX (y));
+CREATE TABLE a (x INT PRIMARY KEY, y INT, z INT, INDEX (y), FAMILY (x, y, z));
 CREATE TABLE b (x INT PRIMARY KEY);
 INSERT INTO a VALUES (1, 1, 1), (2, 1, 1), (3, 2, 2), (4, 2, 2);
 INSERT INTO b VALUES (1), (2), (3), (4);

